ReactiveCocoa/Swift/Property.swift

```diff
@@ -112,18 +112,24 @@ public final class MutableProperty<Value>: MutablePropertyType {
 
 	/// A signal that will send the property's changes over time,
 	/// then complete when the property has deinitialized.
-	public lazy var signal: Signal<Value, NoError> = { [unowned self] in
-		var extractedSignal: Signal<Value, NoError>!
-		self.producer.startWithSignal { signal, _ in
-			extractedSignal = signal
-		}
-		return extractedSignal
-	}()
+	public let signal: Signal<Value, NoError>
 
 	/// A producer for Signals that will send the property's current value,
 	/// followed by all changes over time, then complete when the property has
 	/// deinitialized.
-	public let producer: SignalProducer<Value, NoError>
+	public lazy var producer: SignalProducer<Value, NoError> = { [weak self] in
+		return SignalProducer { observer, disposable in
+			guard let property = self else {
+				observer.sendCompleted()
+				return
+			}
+		
+			disposable += property.withValue { value in
```

Comment:
This was implemented this way on purpose: the source of truth should be the `producer`, not the `signal`.